test/integration/master/crd_test.go

```diff
@@ -160,6 +160,7 @@ func TestCRDOpenAPI(t *testing.T) {
 			},
 			Validation: &apiextensionsv1beta1.CustomResourceValidation{
 				OpenAPIV3Schema: &apiextensionsv1beta1.JSONSchemaProps{
+					Type: "object",
```

Comment:
is the root type still optional? or is there any new validation?

Reply:
it's needed now for structural schemas, yes. We require structural schemas for publishing.